packages/data-context/src/sources/MigrationDataSource.ts

```diff
@@ -51,11 +65,17 @@ export class MigrationDataSource {
 
   async initialize () {
     this._config = null
+    const config = await this.parseCypressConfig()
+
     await this.initializeFlags()
-    this.filteredSteps = MIGRATION_STEPS.filter((step) => this.shouldShowStep(step))
-    if (this.filteredSteps[0]) {
-      this.setStep(this.filteredSteps[0])
+
+    this.filteredSteps = getStepsForMigration(config)
+
+    if (!this.filteredSteps[0]) {
+      throw Error(`No steps for project! This is impossible.`)
```

Comment:
Why is this impossible?

Reply:
When the predicates (should this step be shown) functions were in the same file, it was easy to make sure that a few steps were always shown. I agree that in this case, throwing an error has very little value. 

Now a better wording would probably be 

```suggestion
    if (!this.filteredSteps[0]) {
      throw Error(`Impossible to initialize a migration. No steps fit the configuration of this project.`)
```

even if it should never happen. If we throw an error, might as well be helpful.

Reply:
@emilyrohrbough looking through the Whimsical workflow, every project has at **least** got the `configFile` migration step - so that's why it's impossible to have a migration flow without any steps.

If we reach an impossible situation, an error seems like a good way to indicate this (mainly during development). That way, if something odd happens, instead of looking at a blank screen and wondering what's going on, you'll have an error.